src/result_set.rs

```diff
@@ -0,0 +1,160 @@
+use crate::{
+    ast::ParameterizedValue,
+    error::Error,
+    transaction::{ColumnNames, ResultRow},
+};
+use std::collections::HashMap;
+
+/// Encapsulates a set of results and their respective column names.
+pub struct ResultSet {
+    pub(crate) rows: Vec<ResultRow>,
+    pub(crate) name_to_index: HashMap<String, usize>,
+}
+
+impl ResultSet {
+    /// Creates a new instance, bound to the given column names and result rows.
+    pub fn new(names: &ColumnNames, values: Vec<ResultRow>) -> ResultSet {
+        ResultSet {
+            name_to_index: Self::build_name_map(names),
+            rows: values,
+        }
+    }
+
+    /// Creates a lookup map for column names.
+    pub(crate) fn build_name_map(names: &ColumnNames) -> HashMap<String, usize> {
+        let mut mapped = HashMap::<String, usize>::new();
+
+        for i in 0..names.names.len() {
+            mapped.insert(names.names[i].clone(), i);
+        }
+
+        mapped
+    }
+
+    /// Creates a new, empty instance.
+    pub(crate) fn empty() -> ResultSet {
+        ResultSet {
+            name_to_index: HashMap::new(),
+            rows: Vec::new(),
+        }
+    }
+
+    /// Finds a column index for a name.
+    pub fn column_index(&self, name: &str) -> Result<usize, Error> {
+        match self.name_to_index.get(name) {
+            None => Err(Error::ColumnNotFound(String::from(name))),
+            Some(idx) => Ok(*idx),
+        }
+    }
+
+    /// Returns an interator over wrapped result rows.
+    pub fn iter(&self) -> impl std::iter::Iterator<Item = ResultRowWithName> {
```

Comment:
Not like this. Implement trait `IntoIterator<Item = ResultRowWithName>` and `Iterator<Item = ResultRowWithName>`.

https://doc.rust-lang.org/std/iter/trait.IntoIterator.html
https://doc.rust-lang.org/std/iter/trait.Iterator.html